internal/provider/resource_locally_signed_cert.go

```diff
@@ -266,6 +268,24 @@ func (r *locallySignedCertResource) Create(ctx context.Context, req resource.Cre
 		return
 	}
 
+	// Check whether the certificate is ready for renewal at the time of creation
+	if newState.EarlyRenewalHours.Value >= newState.ValidityPeriodHours.Value && newState.ValidityPeriodHours.Value > 0 {
+		tflog.Warn(ctx, "Certificate is ready for renewal at time of creation")
+		res.Diagnostics.AddWarning(
+			"Certificate is ready for renewal at time of creation",
+			fmt.Sprintf("Early renewal hours (%d) is greater than or equal to validity period hours (%d)", newState.EarlyRenewalHours.Value, newState.ValidityPeriodHours.Value),
+		)
```

Comment:
Was there a specific ask for this warning diagnostic? Practitioners may be intentionally setting their configurations so certificates are recreated every run. Otherwise, we will need an opt-out mechanism, since there is nothing like that available in core.

Reply:
https://hashicorp.slack.com/archives/C02195DNP2B/p1664812872632569?thread_ts=1663791189.094599&cid=C02195DNP2B

Reply:
Created a followup issue for this, since we should consider whether it should happen now, in the future, or differently by enforcing >= 1 instead of 0: https://github.com/hashicorp/terraform-provider-tls/issues/282 👍

Reply:
Thanks for creating the follow-up issue. I've removed the warning diagnostics for now.